packages/xrpl/src/models/transactions/permissionedDomainSet.ts

```diff
@@ -0,0 +1,48 @@
+import { AuthorizeCredential } from '../common'
+
+import {
+  BaseTransaction,
+  isString,
+  validateBaseTransaction,
+  validateOptionalField,
+  validateRequiredField,
+  validateCredentialsList,
+} from './common'
+
+const ACCEPTED_CREDENTIALS_MAX_LENGTH = 10
+
+export interface PermissionedDomainSet extends BaseTransaction {
+  TransactionType: 'PermissionedDomainSet'
+
+  DomainID?: string
+  AcceptedCredentials: AuthorizeCredential[]
+}
+
+/**
+ * Validate a PermissionedDomainSet transaction.
+ *
+ * @param tx - The transaction to validate.
+ * @throws {ValidationError} When the transaction is invalid.
+ */
+export function validatePermissionedDomainSet(
+  tx: Record<string, unknown>,
+): void {
+  validateBaseTransaction(tx)
+
+  validateOptionalField(tx, 'DomainID', isString)
+  validateRequiredField(
+    tx,
+    'AcceptedCredentials',
+    () => tx.AcceptedCredentials instanceof Array,
+  )
```

Comment:
```suggestion
  validateRequiredField(
    tx,
    'AcceptedCredentials',
    () => Array.isArray(tx.AcceptedCredentials),
  )
```
You could also add `isArray` to the list of helpers in `common.ts`

Reply:
I considered this suggestion, but I didn't fully understand the differences. Are our code-snippets very different?

Reply:
`Array.isArray` is better practice - https://stackoverflow.com/questions/28779255/is-instanceof-array-better-than-isarray-in-javascript

Reply:
Hmm, thanks for the above article.

I don't imagine `PDSet` transaction will be invoked across multiple windows in the browser. I expect transaction submission to complete in a single window. This is the edge case where `instanceof` throws false negatives.

Here is the original blog post: http://blog.niftysnippets.org/2010/09/say-what.html#instanceof

Reply:
Regardless, let's use best practices, because xrpl.js is used in the browser, and we should be avoiding things that may result in edge cases (especially when it's not very difficult to make that change).